global/source-zaza/charmcraft-build-lock-file.yaml

```diff
@@ -0,0 +1,37 @@
+type: charm
+
+parts:
+  charm:
+    build-packages:
+      - tox
+      - git
+      - python3-dev
+      - libffi-dev
+      - libssl-dev
+      - libxml2-dev
+      - libxslt1-dev
+      - libmysqlclient-dev  # for executable mysql_shell 
+    override-build: |
+      apt-get install ca-certificates -y
+      tox -e add-build-lock-file
+    override-stage: |
+      echo "Copying charm to staging area: $CHARMCRAFT_STAGE"
+      NAME=$(ls $CHARMCRAFT_PART_BUILD/build/builds)
+      cp -r $CHARMCRAFT_PART_BUILD/build/builds/$NAME/* $CHARMCRAFT_STAGE/
+      cp $CHARMCRAFT_PART_BUILD/src/build.lock $CHARMCRAFT_STAGE/build.lock
+    override-prime: |
+      # For some reason, the normal priming chokes on the fact that there's a
```

Comment:
is there a bug for this issue?, if there is we should include the reference here so we can know when this can be removed.

Reply:
I think this is part of a bigger piece of work to use the reactive plugin instead of overriding the parts like this.  I guess it's in support of charmcraft 1.6+.  Bug opened: #226